binder/src/main/java/io/grpc/binder/internal/BinderServer.java

```diff
@@ -165,6 +174,28 @@ public synchronized boolean handleTransaction(int code, Parcel parcel) {
     return false;
   }
 
+  static class GoAwayHandler implements LeakSafeOneWayBinder.TransactionHandler {
+    static final GoAwayHandler INSTANCE = new GoAwayHandler();
+
+    @Override
+    public boolean handleTransaction(int code, Parcel parcel) {
+      if (code == BinderTransport.SETUP_TRANSPORT) {
+        int version = parcel.readInt();
+        if (version >= BinderTransport.EARLIEST_SUPPORTED_WIRE_FORMAT_VERSION) {
+          IBinder callbackBinder = parcel.readStrongBinder();
+          try (ParcelHolder goAwayReply = ParcelHolder.obtain()) {
+            // Send empty flags to avoid a memory leak linked to empty parcels (b/207778694).
+            goAwayReply.get().writeInt(0);
+            callbackBinder.transact(SHUTDOWN_TRANSPORT, goAwayReply.get(), null, FLAG_ONEWAY);
+          } catch (RemoteException re) {
+            // Ignore.
```

Comment:
Should it be at least logged?

Reply:
IDK, this situation wasn't logged before and there are already quite a few situations where a  transaction is silently ignored (unrecognized code, wrong wire format version, malformed parcel contents, ...). Is there a reason the post-shutdown case is special enough to log? Or do you feel that the behavior change in this PR has now made it special enough to log?

Reply:
> Or do you feel that the behavior change in this PR has now made it special enough to log?

We are introducing a new transaction (for code `SHUTDOWN_TRANSPORT`), aren't we? I think that gives us some room to rethink this call.

If we don't surface this error at all, buggy tests could be silently passing when they shouldn't and without any indication of issues (which is sometimes prevent by `ExpectedLogMessages` rules).

See go/java-practices/exceptions#empty_catch

Reply:
Actually no, SHUTDOWN_TRANSPORT already exists. This PR is so small because clients already understand it and do the right thing in response.

Reply:
Added logging.